dracut/30ignition/coreos-teardown-initramfs-network.sh

```diff
@@ -39,6 +41,42 @@ propagate_initramfs_networking() {
     fi
 }
 
+# Propagate the ip= karg hostname if desired. The policy here is:
+#
+#     - IF a hostname is specified in static networking ip= kargs
+#     - AND no hostname was set via Ignition (realroot `/etc/hostname`)
+#     - THEN we make the last hostname specified in an ip= karg apply
+#       permanently by writing it into `/etc/hostname`
+#
+# This may no longer be needed when the following bug is fixed:
+# https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/419
+propagate_initramfs_hostname() {
+    if [ -e '/sysroot/etc/hostname' ]; then
+        echo "info: hostname is defined in the real root"
+        echo "info: will not attempt to propagate initramfs hostname"
+        return 0
+    fi
+    # Detect if any hostname was provided via static ip= kargs
+    # run in a subshell so we don't pollute our environment
+    hostname=$(
+        last_nonempty_hostname=''
+        # Inspired from ifup.sh from the 40network dracut module. Note that
+        # $hostname from ip_to_var will only be nonempty for static networking.
+        for iparg in $(getargs ip=); do
+            ip_to_var $iparg
+            [ -n "$hostname" ] && last_nonempty_hostname="$hostname"
+        done
+        echo -n "$last_nonempty_hostname"
```

Comment:
I'm sure this works, but without bringing up the source for `ip_to_var` and digging in, it's not obvious that nothing will get emitted from this subshell other than the name. Can we just `echo "$last_nonempty_hostname" > my-hostname` here and then source that in the main shell?

Reply:
Good point. I've updated it to use a temporary file.